html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+
+In the above example we animated the CSS `background-color` property. While there are plenty of CSS properties you can animate, you can't animate them all. So keep [this list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties) of "CSS animatable" properties handy so you can check whether or not you can do the transition you want.
+
+Now let's make a basic transition. There are two ways of writing a transition. One is the long form and the other is the short form. The long form looks like this:
+
+~~~css
+button {
+  /* ... other css properties ... */
+  background-color: white;
+  transition-property: background-color;
+  transition-duration: 2s;
+  transition-timing-function: ease-out;
+  transition-delay: 1s;
+}
+
+button:hover {
+  background-color: black;
+}
+~~~
+
+The `transition-property` tells us what CSS property will be animated. In this case it is the `background-color` which will change from white to black. The `transition-duration` means the fade from white to black will gradually happen over 2 seconds. The `transition-timing-function` lets us change the speed of the transition over the course of its duration. Here it will `ease-out`, meaning the color change will be faster at the start than at the end of the transition. The `transition-delay` means the transition will start 1 second after the cursor rests on the button. The short form of this is:
+
+~~~css
+button {
+  /* ... other css properties ... */
+  background-color: white;
+  transition: background-color 2s ease-out 1s;
+}
+
+button:hover {
+  background-color: black;
+}
+~~~
+
+The above transition will occur when the user hovers their mouse over the button as indicated by the `:hover` pseudo-class. Different pseudo-classes may be used such as `:active` or `:focus` and you can find the [full list here](https://developer.mozifont-sizella.org/en-US/docs/Web/CSS/Pseudo-classes). We will now explore each of those pseudo-classes one by one.
+
+Just kidding! A skim over that list to familiarize yourself with the CSS pseudo-classes will be enough. While you're there, you absolutely should read the MDN article for [using CSS transitions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions). Make sure to pay attention to the note about the `auto` value and some of the finer points around using transition on multiple properties at the same time. 
+
+### Optimization
+
+Optimization is a subject that can lead down quite a few rabbit-holes and is generally not worth focusing on at this intermediate stage. However there are a couple of things you need to keep in mind regarding CSS transitions and animations in general. Feel free to do your own research but if you've started to read about GPU threads regarding this particular topic then you've gone too far, although it is certainly interesting!
```

Comment:
This statement is generally true, but not really applicable to CSS.  There just isn't that much optimization to think and worry about in the first place.  Also, I'd name this section "Performance" and use that term throughout instead of 'optimization'